pop_tools/budget.py

```diff
@@ -0,0 +1,334 @@
+        )
+        raise IOError(error_msg)
+
+    # Drop coordinates for shift, roll, etc.
+    coord_vars = [c for c in ds.coords if c != 'time']
+    ds = ds.drop(coord_vars)
+    if 'z_w_top' in ds.dims:
+        ds = ds.drop('z_w_top')
+        ds = ds.rename({'z_w_top': 'z_t'})
+    if 'z_w_bot' in ds.dims:
+        ds = ds.drop('z_w_bot')
+        ds = ds.rename({'z_w_bot': 'z_t'})
+
+    # Force datetime
+    if not np.issubdtype(ds.time.dtype, np.datetime64):
+        raise TypeError('Please input a dataset with a time coordinate of type datetime.')
+
+    # Force chunking.
+    if not ds.chunks:
+        raise IOError('Please input a dataset with chunks (i.e., dask arrays).')
+    return ds
+
+
+def regional_tracer_budget(
+    ds, grid, mask=None, mask_int=None, budget_depth=None, sum_depth=True, sum_area=True
+):
+    """Return a regional tracer budget on the POP grid.
+
+    Parameters
+    ----------
+    ds : `xarray.Dataset`
+      Dataset containing global POP output, with the tracer suffix removed:
+      * UE
+      * VN
+      * WT
+      * HDIFE
+      * HDIFN
+      * HDIFB
+      * DIA_IMPVF
+      * KPP_SRC
+      * FvICE (optional)
+      * FvPER (optional)
+      * STF (optional; rename from FG_{tracer} for CO2)
+      * SMS (optional; rename from J_{tracer})
+    grid : str
+      POP grid (e.g., POP_gx3v7, POP_gx1v7, POP_tx0.1v3)
+    mask : `xarray.DataArray`, optional
+      Mask on POP grid with integers for region of interest. If None, use REGION_MASK.
+    mask_int : int
+      Number corresponding to integer on mask. E.g., 1 for the Southern Ocean for REGION_MASK.
+    budget_depth : int, optional
+      Depth to compute budget to in m. If None, compute for full depth.
+    sum_depth : bool, optional
+      If True, sum across z_t to return a budget integrated over depth.
+    sum_area : bool, optional
+      If True, sum across nlat/nlon to return a budget integrated over area.
+
+    Returns
+    -------
+    reg_budget: `xarray.Dataset`
+      Dataset containing integrated budget terms over masked POP volume.
+    """
+    ds = _process_input_dataset(ds)
+    grid = _process_grid(grid)
+    if mask is None:
+        # Default to REGION_MASK from POP.
+        mask = grid.mask
+    if mask_int is None:
+        raise ValueError('Please supply an integer for your mask via the `mask_int` keyword.')
+    mask = mask == mask_int
+    mask = mask.drop(mask.coords)
+
+    # Compute maximum k-index for budget.
+    if budget_depth is not None:
+        # Convert from m to cm.
+        budget_depth *= 100
+        kmax = _compute_kmax(budget_depth, grid.z)
+    else:
+        kmax = None
+
+    ladv = _compute_lateral_advection(ds, grid, mask, kmax=kmax)
+    vadv = _compute_vertical_advection(ds, grid, mask, kmax=kmax)
+    lmix = _compute_lateral_mixing(ds, grid, mask, kmax=kmax)
+    vmix = _compute_vertical_mixing(ds, grid, mask, kmax=kmax)
+    reg_budget = xr.merge([ladv, vadv, lmix, vmix])
```

Comment:
If you were to use xgcm, you could replace the above with something like
```python
grid = xgcm.Grid(grid_ds)
ladv = -grid.diff(adv_flux_x, 'X') - grid.diff(adv_flux_y, 'Y')
vadv = -grid.diff(adv_flux_z, 'Z')
```
etc.